File: contentproviderdemo/app/src/main/java/com/example/contentproviderdemo/MainActivity.java
package com.example.contentproviderdemo;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;

import androidx.appcompat.app.AppCompatActivity;

public class MainActivity extends AppCompatActivity implements View.OnClickListener {
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        init();
    }

    @Override
    public void onClick(View view) {
        int viewId = view.getId();
        Intent intent = new Intent();

        if (viewId == R.id.btn_simple_content_provider) {
            intent.setClass(this, SimpleContentProviderActivity.class);
        } else if (viewId == R.id.btn_custom_content_provider) {
            intent.setClass(this, CustomContentProviderActivity.class);
        }

        startActivity(intent);
    }

    private void init() {
        findViewById(R.id.btn_simple_content_provider).setOnClickListener(this);
        findViewById(R.id.btn_custom_content_provider).setOnClickListener(this);
    }
}

File: recyclerviewdemo/app/src/main/java/com/example/recyclerviewdemo/activity/MainActivity.kt
package com.example.recyclerviewdemo.activity

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import com.example.recyclerviewdemo.R
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        btnVerticalList.setOnClickListener {
            startActivity(Intent(this, VerticalListActivity::class.java))
        }

        btnGridList.setOnClickListener {
            startActivity(Intent(this, GridListActivity::class.java))
        }
    }
}